### PR TITLE
Add Cosmos DB For PostgreSQL beta sdk and tests 

### PR DESCRIPTION
**Pull request by senatol**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by senatol** [comment]
This will be our first SDK package release so we will release them as beta version for at least 1 month before releasing the full version.
namespace approval: https://github.com/Azure/azure-sdk/issues/6002

**Comment by senatol** [comment]
Link for ADO work item created after running prepare release script: https://dev.azure.com/azure-sdk/Release/_workitems/edit/15928/

**Comment by live1206** [comment]
@senatol Please re-record the tests locally for the failing ones, details can be found via https://dev.azure.com/azure-sdk/public/_build/results?buildId=2814548&view=ms.vss-test-web.build-test-results-tab.

**Comment by live1206** [comment]
```
cspell run complete, restoring original configuration and removing temp file.
./sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:23:23 - Unknown word (Citus)
./sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:118:23 - Unknown word (Citus)
./sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:394:635 - Unknown word (citus)
./sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:395:923 - Unknown word (citus)
./sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:413:159 - Unknown word (Groupsv)
./sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:428:23 - Unknown word (Citus)
##[error]Spelling errors detected. To correct false positives or learn about spell checking see: https://aka.ms/azsdk/engsys/spellcheck
```
Please bypass the spell check in https://github.com/Azure/azure-sdk-for-net/blob/main/.vscode/cspell.json if you think they are valid.

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/src/autorest.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  ConfigurationDataType: CosmosDBForPostgreSqlConfigurationDataType
+  CheckNameAvailabilityResourceType: CosmosDBForPostgreSqlCheckNameAvailabilityResourceType
+  ServerRole: CosmosDBForPostgreSqlServerRole
+  RoleListResult: CosmosDBForPostgreSqlRoleListResult
+  ServerRoleGroupConfiguration: CosmosDBForPostgreSqlServerRoleGroupConfiguration
+  SimplePrivateEndpointConnection: CosmosDBForPostgreSqlSimplePrivateEndpointConnection
+  ServerConfigurationData: CosmosDBForPostgreSqlServerConfigurationData
+
+prepend-rp-prefix:
+- Cluster
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/src/autorest.md:65]: Did you re-export API after this change? I can still see `Cluster` in the API file

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.CosmosDBForPostgreSql.ClusterResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.CosmosDBForPostgreSql.ClusterResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class ClusterData : Azure.ResourceManager.Models.TrackedResourceData
+    {
+        public ClusterData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
+        public string AdministratorLogin { get { throw null; } }
+        public string AdministratorLoginPassword { get { throw null; } set { } }
+        public string CitusVersion { get { throw null; } set { } }
+        public bool? CoordinatorEnablePublicIPAccess { get { throw null; } set { } }
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:24]: For `bool` type properties, consider to rename them to `IsXXXEnabled`.

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+        public int? NodeStorageQuotaInMb { get { throw null; } set { } }
+        public int? NodeVCores { get { throw null; } set { } }
+        public System.DateTimeOffset? PointInTimeUTC { get { throw null; } set { } }
+        public string PostgresqlVersion { get { throw null; } set { } }
+        public string PreferredPrimaryZone { get { throw null; } set { } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.CosmosDBForPostgreSql.Models.SimplePrivateEndpointConnection> PrivateEndpointConnections { get { throw null; } }
+        public string ProvisioningState { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<string> ReadReplicas { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.CosmosDBForPostgreSql.Models.ServerNameItem> ServerNames { get { throw null; } }
+        public string SourceLocation { get { throw null; } set { } }
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:44]: Is this type `AzureLocation`?

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+        public string SourceResourceId { get { throw null; } set { } }
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:45]: Is this type `ResourceIdentifier`?

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+        public int? StartHour { get { throw null; } set { } }
+        public int? StartMinute { get { throw null; } set { } }
+    }
+    public partial class NameAvailability
+    {
+        internal NameAvailability() { }
+        public string Message { get { throw null; } }
+        public string Name { get { throw null; } }
+        public bool? NameAvailable { get { throw null; } }
+        public string ResourceType { get { throw null; } }
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs:525]: Is this type `Azure.Core.ResourceType`?

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/src/autorest.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+rename-mapping:
+  ClusterListResult: CosmosDBForPostgreSqlClusterListResult
+  ClusterServerListResult: CosmosDBForPostgreSqlClusterServerListResult
+  NameAvailability: CosmosDBForPostgreSqlNameAvailability
+  FirewallRuleListResult: CosmosDBForPostgreSqlFirewallRuleListResult
+  MaintenanceWindow: CosmosDBForPostgreSqlMaintenanceWindow
+  ClusterConfigurationListResult: CosmosDBForPostgreSqlClusterConfigurationListResult
+  ConfigurationDataType: CosmosDBForPostgreSqlConfigurationDataType
+  CheckNameAvailabilityResourceType: CosmosDBForPostgreSqlCheckNameAvailabilityResourceType
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/src/autorest.md:57]:
```
`CheckNameAvailability` operation:  

The method name should be `Check[Resource/RP name]NameAvailability`  

The parameter / response model name should be `[Resource/RP name]NameAvailabilityXXX`  

If there is a `Reason` property which is used to represent the unavailable reason code, the name should be `[Resource/RP name]NameUnavailableReason` 
```

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/src/autorest.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  WAN: Wan
+  Ipv4: IPv4|ipv4
+  Ipv6: IPv6|ipv6
+  Ipsec: IPsec|ipsec
+  SSO: Sso
+  URI: Uri
+  Etag: ETag|etag
+
+rename-mapping:
+  ClusterListResult: CosmosDBForPostgreSqlClusterListResult
```
Comment (live1206) [inline sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/src/autorest.md:50]: For the ones prepending RP prefix, consider to just put them under section `prepend-rp-prefix`

**Review by live1206 — COMMENTED** [review]
@senatol 
Really appreciate your effort on the polishing type naming in .NET SDK! Even we don't normally
require them for beta versions before first GA. More details can be found via https://github.com/Azure/autorest.csharp/blob/feature/v3/docs/mgmt/polishing.md

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs (hunk 3 above)
Comment (senatol) [reply sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs]: Updated type to AzureLocation

File: sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs (hunk 4 above)
Comment (senatol) [reply sdk/cosmosdbforpostgresql/Azure.ResourceManager.CosmosDBForPostgreSql/api/Azure.ResourceManager.CosmosDBForPostgreSql.netstandard2.0.cs]: Updated type to ResourceIdentifier

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**